src/create-with-theme.js

```diff
@@ -1,31 +1,26 @@
 const React = require('react');
-const PropTypes = require('prop-types');
 const channel = require('./channel');
+const createThemeListener = require('./create-theme-listener');
 
 function createWithTheme(CHANNEL = channel) {
+  const themeListener = createThemeListener(CHANNEL);
   return Component =>
     class withTheme extends React.Component {
-      static contextTypes = {
-        [CHANNEL]: PropTypes.func,
-      };
+      static contextTypes = themeListener.contextTypes;
+      constructor(props) {
+        super(props);
+        this.themeSubscribe = themeListener.subscribe.bind(this);
```

Comment:
createThemeListener() returns basically a mixin, it would be more explicit to merge that object with the prototype of withTheme class.